### DISCUSSION TITLE
Code structure

### DISCUSSION THREAD

**Discussion by saschagu** [biologist79/ESPuino#62] [General]
Hi all!
I really love this project and the great work you have done! I would also like to contribute, but find it difficult to read the code, if it is all in one source file with nearly 5000 LOC's. Do you have any plans to restructure the code in classes or at least several source/header files, for example for topics like web server, MQTT, playlist control etc.?
I think this would make the code more readable and configuration with compiler flags more easy. I know its initially a lot of work without new features and potentially new bugs, but I think if more people start working on/with the code it would be well invested time.
Thanks again for your work!

**Comment by biologist79** [comment]
Hi,

yes, I do have plans to restructure the code in several source/header-files. Guess I'll do that within the next days; stay tuned :-)

**Reply by saschagu** [reply]
Great, thanks!

**Reply by biologist79** [reply]
Has been solved and the new code-structure has become the new master-branch a few weeks ago.
